File: Dynamic Programming/55_jump_game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        int dest = n - 1;

        for(int i = dest - 1; i >= 0; i -= 1) {
            if(i + nums[i] >= dest) dest = i;
        }

        return dest == 0;
    }
};

// Revised - 01/06/2025

File: Dynamic Programming/55_jump_game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        dest = len(nums) - 1
        for i in range(dest - 1, -1, -1):
            if i + nums[i] >= dest:
                dest = i
        return dest == 0
    
# revised - 02/19/2025